src/sentry/replays/consumers/recording_buffered.py

```diff
@@ -210,14 +219,36 @@ def process_message(buffer: RecordingBuffer, message: bytes) -> None:
     # Append an upload event to the state object for later processing.
     buffer.upload_events.append(
         {
-            "payload": recording_data,
-            "project_id": decoded_message["project_id"],
-            "replay_id": decoded_message["replay_id"],
-            "retention_days": decoded_message["retention_days"],
-            "segment_id": headers["segment_id"],
+            "key": make_recording_filename(
+                project_id=decoded_message["project_id"],
+                replay_id=decoded_message["replay_id"],
+                retention_days=decoded_message["retention_days"],
+                segment_id=headers["segment_id"],
+            ),
+            "value": recording_data,
         }
     )
 
+    if replay_video := decoded_message.get("replay_video"):
+        assert isinstance(replay_video, bytes)
+        buffer.upload_events.append(
+            {
+                "key": make_video_filename(
+                    project_id=decoded_message["project_id"],
+                    replay_id=decoded_message["replay_id"],
+                    retention_days=decoded_message["retention_days"],
+                    segment_id=headers["segment_id"],
+                ),
+                "value": replay_video,
+            }
+        )
+
+        # For now we only publish replay-events if they were part of the video payload.
+        # Otherwise we would double publish.  In the future this can be removed once the
+        # recording consumer is responsible for publishing all replay-events to snuba.
+        if replay_event := decoded_message.get("replay_event"):
+            buffer.replay_events.append(replay_event)
```

Comment:
For either consumer type, I'm only publishing the replay-event if the payload also contained the replay-video attribute.  I don't want to double publish on non-replay-video types.